HIP/hip-657.md

```diff
@@ -0,0 +1,115 @@
+
+
+To address this issue, the proposed solution is to create a new key type called the METADATA key. This key will allow NFTs in a collection to update their metadata.
+
+
+The use of the METADATA key to update NFT metadata will not affect the functionality of existing NFTs on the Hedera Network. It will be available for new and existing NFTs.
+
+
+## User stories
+
+
+- As a collector of NFTs, I want to be able to update the metadata of an NFT in my collection so that I can add new information or make changes to the existing metadata.
+
+
+- As a creator of NFTs, I want to be able to update the metadata of an NFT while leaving an auditable on-chain history of when the update happened and which key was used to sign for the update action so that I can retain as much transparency as possible. 
+
+
+- As a creator of NFTs, I want to use IPFS to manage my dynamic NFT’s metadata so that I don’t have to use a centralized file storage solution to enable the ability to edit.  
+
+
+- As a developer building applications on top of the Hedera Network, I want the ability to update the metadata of NFTs so that I can create dynamic NFTs and build more advanced use cases.
+
+
+## Specification
+
+
+The METADATA key will be a new key that allows for updating the metadata of an NFT in a new NftMetadataUpdate transaction.
+
+The NftMetadataUpdate transaction should accept the following parameters:
+	- identifier (string): the 0.0.<token-type>.<serial-no>
+- the new NFT metadata: the new metadata for the NFT
+
+
+The NftMetadataUpdate transaction should have a new token type field called "dynamic_metadata" of type bool, which is valid only for non-fungible tokens. By default, this field is set to false (meaning the metadata for this token type is immutable). If set to true, owners of this token type can change their NFT's metadata via the NftMetadataUpdate transaction.
+
+
+Example transaction:
+```js
+NftMetadataUpdate()
+     .setTokenId(tokenId)
+     .setMetaData(newMetaData)
+     .sign(METADATAKEY)
+     .execute(client);
+```
+
+
+Requirements: The NFT must exist on the Hedera network and should only be allowed to execute if it is signed by the new METADATAKEY. The new metadata must be a valid string. The update must be made atomically and be recorded on the Hedera network.
+
+## Backwards Compatibility
+
+
+The addition of the METADATA key will not affect the functionality of existing NFTs on the Hedera Network because it will be available for use on new and existing NFTs.
+
+## Security Implications
+
+
+The use of the METADATA key and creation of a NftMetadataUpdate transaction will not introduce any new security concerns.
+
+## How to Teach This
+TBD
+
+
+
+## Reference Implementation
+
+
+The reference implementation must be complete before any HIP is given the status of “Final”. The final implementation must include test code and documentation.
+
+
+## Rejected Ideas
+
+
+## Open Issues
+
+
+There are currently no open issues with this proposal.
```

Comment:
Q's

1. Can token creators make such an update when they do not own the NFT?
2. Can token creators set the NFT to be immutable/static in a way that future owners cannot make them dynamic?
3. Is the key type applicable at the token class of the individual NFT level?

Reply:
> Can token creators make such an update when they do not own the NFT?
Yes, absolutely. The account which controls the private key associated with the METADATA public key can update the METADATA field on any NFT within the Collection (token_id) independently from which account currently custodies the NFT asset.

Reply:
> Can token creators set the NFT to be immutable/static in a way that future owners cannot make them dynamic?

Similar to all other keys on a token_id, if it is not set at the time of token creation, then the NFT Collection (token_id) is considered immutable whereas a new key cannot be added. This is consistent with all other keys on the token_id. 

Depending on the outcome of [HIP-540](https://github.com/hashgraph/hedera-improvement-proposal/pull/540), the METADATA key can remove itself thus entering it into the state of "no METADATA key defined" which also is considered immutable since no key can be re-added once removed (or not initially set).

Reply:
> Is the key type applicable at the token class of the individual NFT level?

The token_id (token class), but has the ability to update the METADATA value for any NFT within that token class.